### PR TITLE
Serve images in next-gen image formats where possible

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-1450](https://trello.com/c/AwCyN4st/1450-page-speed-improvements)

### Context

We use basic JPEG/PNG images for the most part; a recommendation is to use newer image formats that enable better compression, which improves the page load speed. Instead of manually saving each image in the various formats, we can generate the different formats during the Docker build process and dynamically serve them when they exist from Rails.

We should be able to build on this with a couple of further optimisations; only loading images 'above the fold' on the first paint and serving device-appropriate images (so smaller images for mobile, for example).

It would be nice to serve background images in the different formats as well, but I'm not sure how possible that is without relying on JS - I'll investigate in another PR.

### Changes proposed in this pull request

- Add NextGenImages class

Introduces a class that replaces `` tags with `<picture>` tags and applicable `<source>` tags for any next-gen images that exist with the same file name as the original image.

This will allow us to generate next-gen images as part of the Docker build process and then Rails will introduce them into the responses dynamically. As most of our pages are static and cached it shouldn't have a big performance overhead.

- Add next-gen images to all HTML responses

Any HTML responses are intercepted and the next-gen images are introduced when they exist.

Enabled only in preprod initially.

- Tweak CSS to support picture instead of img tags

Some of the CSS targets a specific `img` tag with the `>` selector; these need updating to target the `img` within the new `picture` tag.

- Correct HTML closing tag

It was getting messed up running through Nokogiri (even though it rendered fine by browsers like this).

### Guidance to review

As it stands the existing images will be wrapped in a `<picture>` tag with a single `<source>` for the original image. When we update the content Docker build process to generate images in next-gen formats it should start serving those as well.

I've tested this locally with a decent amount of clicking about and it doesn't seem to break anything. Its only enabled in preprod for now so that I can get some other people to give it a decent smoke test.

### CODE REVIEW

File: lib/next_gen_images.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      source.set_attribute("type", mime_type(src))
+    end
+  end
+
+  def mime_type(src)
+    ext = File.extname(src)
+
+    case ext
+    when ".webp"
+      "image/webp"
```
Comment (peteryates) [inline lib/next_gen_images.rb:50]: Is there a reason why `.webp` is a special case here (vs `.jp2`)?

Comment (ethax-ross) [reply lib/next_gen_images.rb:50]: Yeah for some reason Rack::Mime doesn't know about webp so gives it the wrong mime type, but jp2 it gets right 🙄

**Review by peteryates — DISMISSED** [review]
Nice. Logic-wise it seems sound. Left a question on the case statement, perhaps it just needs a small comment to explain.

**Review by peteryates — APPROVED** [review]
Nice! 🖼️

**Pull request merged**